packages/plugin-core/src/ExtensionProvider.ts

```diff
@@ -26,6 +26,10 @@ export class ExtensionProvider {
     return ExtensionProvider.extension;
   }
 
+  static getState() {
+    return ExtensionProvider.extension.getState();
```

Comment:
instead of exposing a method like getState, can we just make it specific to getInlineNoteRefs to enable this feature?  This is moving us back towards a global state pattern.

Even better if we can make the inlineNoteRef data non-global.